Source/SpacePark-API/DataAccess/APICollector.cs

```diff
@@ -0,0 +1,309 @@
+                httpclient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                var httpresponse = httpclient.GetAsync(url).Result;
+                httpresponse.EnsureSuccessStatusCode();
+                jsonResult = httpresponse.Content.ReadAsStringAsync().Result;
+            }
+
+            if (!string.IsNullOrWhiteSpace(jsonResult))
+            {
+                var parsedShip = JsonConvert.DeserializeObject<SpaceShip>(jsonResult);
+                var resultobject = JsonConvert.DeserializeObject<JObject>(jsonResult);
+                foreach (var property in resultobject.Properties())
+                    if (property.Name == "length")
+                        parsedShip.ShipLength = property.Value.ToString().Replace(",", "").Replace(".", "");
+                return parsedShip;
+            }
+
+            throw new Exception("Parse was empty; Is the URL in correct format?");
+        }
+
+        #endregion
+
+        public static Person ParseUser(string name)
+        {
+            name = System.Web.HttpUtility.UrlDecode(name); //Decode URL if it contains spacebar. This is to ensure it is always correct
+            //-no matter what encoding is used. Most common is percent20-encoding, but could also be plus-encoding(so that spacebars-
+            //will be replaced with + signs)
+            var foundUser = false;
+            var user = new Person();
+            for (var i = 1; i <= 9; i++)
+                if (!foundUser)
+                {
+                    var tempUsers =
+                        ReturnUsersFromList("https://swapi.dev/api/people/?page=" + i);
+                    foreach (var s in tempUsers)
+                    {
+                        if (s.Name.ToLower().Replace(" ", "").Replace("-", "") != name.ToLower().Replace(" ","").Replace("-","")) continue;
+                        foundUser = true;
+                        user = s;
+                        break;
+                    }
+                }
+                else
+                {
+                    break;
+                }
+
+            return foundUser ? user : null;
+        }
+
+        public static Person ParseUserAsync(string name) //This contains a threadstart for the private corresponding method
+        {
+            var user = new Person();
+            var thread = new Thread(() => { user = ParseUser(name); });
+            thread.Start();
+            thread.Join(); //By doing join it will wait for the method to finish
+            return user;
+        }
+
+        public static SpaceShip ParseShip(string model)
+        {
+            model = System.Web.HttpUtility.HtmlDecode(model);
+            
+            var foundShip = false; //A check to verify a ship has been found by name
+            var
+                ship = new SpaceShip(); //Had to initialize because code didn't realize it was bound to the foundShip bool and i got build errors.
```

Comment:
From what I have read its not best practice to put comments on the end of the line... I would suggest in a new line, reasons for this is that sometimes you have to scroll horizontally if the comment is too long, or perhaps the code is too long so then the comment is just missed off the side.

Also I think its good to write comments for your fellow programmers, but, perhaps when turning in the project clean out the unnecessary comments, like the ones the one that has  foundShip, the variable alone should be enough to tell you what its doing :>

Couldn't find anything official but here is a nice blog that gives some insight
[How to comment..](https://www.elegantthemes.com/blog/wordpress/how-to-comment-your-code-like-a-pro-best-practices-and-good-habits)